### PR TITLE
docs: update branch filtering documentation

### PR DESCRIPTION
**Pull request by dwmarshall**
The first question I have when I see "regular expression" is "what
flavor," so I have made that explicit. Plus, I have a thing about
regular expressions that end with .*$ or .*, so I had to fix that.

### CODE REVIEW

File: docs/user-guide/configuration/workflow.md [hunk 1]
```diff
@@ -119,10 +119,10 @@ If job names are prefixed with tildes in a `requires` line, then the job will st
...
+To trigger jobs in your pipeline after a specific branch is committed, you can use branch filtering. The format is `~commit:branchName` or `~pr:branchName`. Branches may also be specified by using a ([JavaScript flavor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)) regular expression (e.g. `~commit/^feature-/`). Note: Flags are not supported.
 
 ### Example
-In the following example, when staging branch is committed, `staging-commit` and `all-commit` are triggered. Also, when this pipeline's branch is committed, `main` and `all-commit` are triggered. When a pull request is opened in staging branch, `staging-pr` is triggered.
+In the following example, when branch `staging` is committed, `main`, `staging-commit` and `all-commit` are triggered. Also, when branch `master` is committed, `main` and `all-commit` are triggered. When a pull request is opened in branch `staging`, `staging-pr` is triggered.
```
Comment (tkyi) [inline docs/user-guide/configuration/workflow.md]: When a commit is made to branch `staging`, `main` should not run.

Comment (dwmarshall) [reply docs/user-guide/configuration/workflow.md]: My bad! Example amended.

File: docs/user-guide/configuration/workflow.md [hunk 2]
```diff
@@ -119,10 +119,10 @@ If job names are prefixed with tildes in a `requires` line, then the job will st
...
-To trigger jobs in your pipeline after a specific branch is committed, you can use branch filtering jobs. The format is `~commit:branchName` or `~pr:branchName`. Also you can use regex filter after `~commit:` or `~pr:` (e.g. `~commit:/^feature-.*$/`).  
+To trigger jobs in your pipeline after a specific branch is committed, you can use branch filtering. The format is `~commit:branchName` or `~pr:branchName`. Branches may also be specified by using a ([JavaScript flavor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)) regular expression (e.g. `~commit/^feature-/`). Note: Flags are not supported.
 
 ### Example
-In the following example, when staging branch is committed, `staging-commit` and `all-commit` are triggered. Also, when this pipeline's branch is committed, `main` and `all-commit` are triggered. When a pull request is opened in staging branch, `staging-pr` is triggered.
+In the following example, when branch `staging` is committed, staging-commit` and `all-commit` are triggered. Also, when branch `master` is committed, `main` and `all-commit` are triggered. When a pull request is opened in branch `staging`, `staging-pr` is triggered.
```
Comment (tkyi) [inline docs/user-guide/configuration/workflow.md]: Missing a tic
```suggestion
In the following example, when branch `staging` is committed, `staging-commit` and `all-commit` are triggered. Also, when branch `master` is committed, `main` and `all-commit` are triggered. When a pull request is opened in branch `staging`, `staging-pr` is triggered.
```

**Comment by dwmarshall** [comment]
Would it be useful to point out that what I have written as a stupid branch filter (`~commit:/./`) might be better characterized as a way to trigger a job whenever any branch gets a commit?

**Comment by tkyi** [comment]
@dwmarshall Yeah, we could update the comment to explain what the regex would be used for

**Review by tkyi — APPROVED** [review]

**Pull request merged**